From: Andi Kleen <ak@linux.intel.com>
Subject: x86: autoload microcode driver on Intel and AMD systems v2
References: http://lists.opensuse.org/opensuse-kernel/2011-11/msg00075.html
Patch-Mainline: 78ff123b05fb15beb1ad670372eea0d299d0b8af undefined


Signed-off-by: Thomas Renninger <trenn@suse.de>

Don't try to describe the actual models for now.

v2: Fix typo: X86_VENDOR_ANY -> X86_FAMILY_ANY (trenn)

Signed-off-by: Andi Kleen <ak@linux.intel.com>
Signed-off-by: Thomas Renninger <trenn@suse.de>
Acked-by: H. Peter Anvin <hpa@zytor.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

Automatically created from "patches.arch/009-x86_autoload.patch" by xen-port-patches.py

--- head.orig/arch/x86/kernel/microcode_core-xen.c	2012-02-09 14:22:00.000000000 +0100
+++ head/arch/x86/kernel/microcode_core-xen.c	2012-02-16 14:36:43.000000000 +0100
@@ -39,6 +39,7 @@
 
 #include <asm/microcode.h>
 #include <asm/processor.h>
+#include <asm/cpu_device_id.h>
 
 MODULE_DESCRIPTION("Microcode Update Driver");
 MODULE_AUTHOR("Tigran Aivazian <tigran@aivazian.fsnet.co.uk>");
@@ -183,6 +184,16 @@ static int request_microcode(const char 
 	return error;
 }
 
+#ifdef MODULE
+/* Autoload on Intel and AMD systems */
+static const struct x86_cpu_id microcode_id[] = {
+	{ X86_VENDOR_INTEL, X86_FAMILY_ANY, X86_MODEL_ANY, },
+	{ X86_VENDOR_AMD, X86_FAMILY_ANY, X86_MODEL_ANY, },
+	{}
+};
+MODULE_DEVICE_TABLE(x86cpu, microcode_id);
+#endif
+
 static int __init microcode_init(void)
 {
 	const struct cpuinfo_x86 *c = &boot_cpu_data;
